app/controllers/application_controller.rb

```diff
@@ -1,9 +1,10 @@
 class ApplicationController < ActionController::Base
+  skip_before_action :verify_authenticity_token
   before_action :store_location
+  
   def store_location
     unless params[:controller] == "devise/sessions"
-      url = root_path
-      session[:user_return_to] = url
+      session[:user_return_to] = root_path unless params[:controller] == "devise/sessions"
```

Comment:
You have this check twice, you can remove leave it to the one liner.

Right now, you have:
```ruby
unless condition
  action unless same_condition
end
```